RIAssigner/data/PandasData.py

```diff
@@ -28,8 +29,14 @@ def retention_times(self) -> Iterable[int]:
     def retention_indices(self):
         if self._carbon_number_index is not None:
             return self._data[self._carbon_number_index] * 10
+        elif "retention_index" in self._data.columns:
+            return self._data["retention_index"]
         raise KeyError("Dataset does not contain retention indices!")
 
     @retention_indices.setter
-    def retention_indices(self, value: Iterable[int]):
-        raise NotImplementedError()
+    def retention_indices(self, values: Iterable[int]):
+        if len(values) == len(self._data):
+            self._ri_position = self._rt_position + 1  
+            self._data.insert(loc=self._ri_position, column="retention_index", value=values)

```

Comment:
Here you should check if the column already exists and if so, only overwrite the values.
Maybe it would be better to do this initialization step in the constructor. If there is a column with RI values, we use the name of it, if not, we add a new one.